app/utilities/supplementary_data_parser.py

```diff
@@ -0,0 +1,88 @@
+from typing import Mapping
+
+from marshmallow import (
+    EXCLUDE,
+    INCLUDE,
+    Schema,
+    ValidationError,
+    fields,
+    validate,
+    validates_schema,
+)
+
+from app.authentication.auth_payload_versions import SupplementaryDataSchemaVersion
+from app.utilities.metadata_parser_v2 import VALIDATORS, StripWhitespaceMixin
+
+
+class ItemsSchema(Schema):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+
+
+class ItemsData(Schema, StripWhitespaceMixin):
+    pass
+
+
+class SupplementaryData(Schema, StripWhitespaceMixin):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+    schema_version = VALIDATORS["string"](
+        validate=validate.OneOf([SupplementaryDataSchemaVersion.V1.value])
+    )
+    items = fields.Nested(ItemsData, required=False, unknown=EXCLUDE)
```

Comment:
Should this be ItemsData or ItemsSchema?.

Reply:
I needed to at ItemsData to allow for when Items is not there. The ItemsSchema class is validating that the identifer key is present when Items exist